plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1222 @@
+func (*JBoss) Description() string {
+	return "Telegraf plugin for gathering metrics from JBoss AS"
+}
+
+func (h *JBoss) checkAuth(host string, uri string) error {
+	url := h.Servers[0]
+
+	method := "POST"
+	req, err := http.NewRequest(method, url, nil)
+	req.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusUnauthorized {
+		log.Printf("Recieved status code '%v' auth skipped\n", resp.StatusCode)
+		return nil
+	}
+	digestParts := digestParts(resp)
+	digestParts["uri"] = uri
+	digestParts["method"] = method
+	digestParts["username"] = h.Username
+	digestParts["password"] = h.Password
+	postData := []byte("{\"address\":[\"\"],\"child-type\":\"host\",\"json.pretty\":1,\"operation\":\"read-children-names\"}")
+	req, err = http.NewRequest(method, url, bytes.NewBuffer(postData))
+	h.Authorization = getDigestAuthrization2(digestParts)
+	req.Header.Set("Authorization", h.Authorization)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err = client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+	log.Printf("D! JBoss HTTP response code: %d \n", resp.StatusCode)
+
+	if resp.StatusCode != http.StatusOK {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			panic(err)
+		}
+		log.Printf("D! JBoss HTTP response body: %s \n", string(body))
+		return nil
+	}
+	return nil
+}
+
+func digestParts(resp *http.Response) map[string]string {
+	result := map[string]string{}
+	if len(resp.Header["Www-Authenticate"]) > 0 {
+		//        wantedHeaders := []string{"nonce", "realm", "qop"}
+		wantedHeaders := []string{"nonce", "realm"}
+		responseHeaders := strings.Split(resp.Header["Www-Authenticate"][0], ",")
+		for _, r := range responseHeaders {
+			for _, w := range wantedHeaders {
+				if strings.Contains(r, w) {
+					result[w] = strings.Split(r, `"`)[1]
+				}
+			}
+		}
+	}
+	return result
+}
+
+func getMD5(text string) string {
+	hasher := md5.New()
+	hasher.Write([]byte(text))
+	return hex.EncodeToString(hasher.Sum(nil))
+}
+
+func getCnonce() string {
+	b := make([]byte, 8)
+	io.ReadFull(rand.Reader, b)
+	return fmt.Sprintf("%x", b)[:16]
+}
+
+func getDigestAuthrization(digestParts map[string]string) string {
+	d := digestParts
+	ha1 := getMD5(d["username"] + ":" + d["realm"] + ":" + d["password"])
+	ha2 := getMD5(d["method"] + ":" + d["uri"])
+	nonceCount := 00000001
+	cnonce := getCnonce()
+	response := getMD5(fmt.Sprintf("%s:%s:%v:%s:%s:%s", ha1, d["nonce"], nonceCount, cnonce, d["qop"], ha2))
+	authorization := fmt.Sprintf(`Digest username="%s", realm="%s", nonce="%s", uri="%s", cnonce="%s", nc="%v", qop="%s", response="%s"`,
+		d["username"], d["realm"], d["nonce"], d["uri"], cnonce, nonceCount, d["qop"], response)
+	return authorization
+}
+
+func getDigestAuthrization2(digestParts map[string]string) string {
```

Comment:
Should be named more descriptively, also looks like this is the only one used, so maybe remove the other implementation?